bundles/org.openhab.binding.dreamscreen/src/main/java/org/openhab/binding/dreamscreen/internal/DreamScreenDatagramServer.java

```diff
@@ -0,0 +1,272 @@
+        if (dreamScreens.isEmpty()) {
+            stopRefreshing();
+        }
+    }
+
+    void shutdown() {
+        dreamScreens.clear();
+        stopRefreshing();
+    }
+
+    private synchronized void ensureRefreshing() throws IOException {
+        if (this.broadcastAddress == null) {
+            this.broadcastAddress = InetAddress.getByName("255.255.255.255");
+        }
+        if (this.refreshing == null) {
+            final DatagramSocket socket = new DatagramSocket(DREAMSCREEN_PORT, this.hostAddress);
+            socket.setReuseAddress(true);
+            this.serverSocket = socket;
+
+            final Thread server = new Thread(this::runServer, "dreamscreen-tv");
+            server.setDaemon(true);
+            server.start();
+
+            final ScheduledExecutorService scheduler = this.scheduler;
+            if (scheduler != null) {
+                this.refreshing = scheduler.scheduleAtFixedRate(this::requestRefresh, 0, 10, TimeUnit.SECONDS);
+            } else {
+                logger.error("No scheduler found to refresh DreamScreen TV");
+            }
+        }
+    }
+
+    private void requestRefresh() {
+        try {
+            broadcast(0xFF, 0x30, 0x01, 0x0A, new byte[0]);
+        } catch (IOException e) {
+            logger.error("Error requesting refresh of state", e);
+        }
+    }
+
+    private synchronized void stopRefreshing() {
+        final DatagramSocket socket = this.serverSocket;
+        if (socket != null) {
+            socket.close();
+        }
+        this.serverSocket = null;
+
+        final ScheduledFuture<?> refreshing = this.refreshing;
+        if (refreshing != null) {
+            refreshing.cancel(true);
+        }
+        this.refreshing = null;
+    }
+
+    @SuppressWarnings("null")
+    private void runServer() {
+        final byte[] buf = new byte[256];
+
+        while (serverSocket != null && !serverSocket.isClosed()) {
+            try {
+                final DatagramPacket data = new DatagramPacket(buf, buf.length);
+                serverSocket.receive(data);
+
+                final int off = data.getOffset();
+                final int len = data.getLength();
+
+                logger.debug("DreamScreen message: {} {}-{}:{}", data.getAddress(), off, len, buf);
+
+                if (isValidStateMsg(buf, off, len) && isDreamScreen(buf, off, len)) {
+                    logger.debug("Received DreamScreen message from {}", data.getAddress());
+                    refreshDreamScreen(buf, off, len, data.getAddress());
+                }
+            } catch (IOException ioe) {
+                logger.error("Error receiving DreamScreen data", ioe);
+            }
+        }
+    }
+
+    void setHostAddress(@Nullable String value) {
+        try {
+            this.hostAddress = InetAddress.getByName(value);
+        } catch (UnknownHostException e) {
+            logger.error("Cannot set host address to {}", value, e);
```

Comment:
If this happens this would probably mean the thing should be in offline status with this message?
```suggestion
            logger.debug("Cannot set host address to {}", value, e);
```

Reply:
changed